airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/MemorySecretPersistence.java

```diff
@@ -26,7 +27,7 @@ public void write(final SecretCoordinate coordinate, final String payload) {
   }
 
   public Map<SecretCoordinate, String> getMap() {
-    return new HashMap<>(secretMap);
+    return Collections.unmodifiableMap(secretMap);
```

Comment:
why make this unmodifiable? I am a little scared of these classes because their signature is the same as a map but it can break at runtime

Reply:
@sherifnada the class `MemorySecretPersistence` is only being used in tests mainly to assert that the correct secret would be written to the persistence. The reason I convert it to unmodifiableMap is to safe guard it from being mutated in any way outside this class. This just gives me more confidence on the assertions. If you feel that we should not use unmodifiableMap, I can change it

Reply:
should we move this class to a test package to ensure we don't accidentally use it in production?

Reply:
I like alex's suggestion here